UserAuth functions (#54)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/user-auth/auth-functions.mdx

---
title: User Auth Functions

> description: Learn about the API functions provided by the Product Details dropin.

Reviewer: ```suggestion
description: Learn about the API functions provided by the User Auth dropin.
```

---

## confirmEmail

> The `confirmEmail` function uses the supplied `customerEmail` and `customerConfirmationKey` parameters to complete the customer activation process. Adobe Commerce sends the confirmation key to the customer when they request to create an account. The function calls the [`confirmEmail` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/confirm-email/).

Reviewer: I've only been linking the name of the query, mutation, or type. It's a trivial concern, but let me know what you prefer so that we're consistent across dropins.

For example:

```
[`confirmEmail](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/confirm-email/) mutation
```

confirmEmail({
  customerEmail: "abc@example.com",
  customerConfirmationKey: "1234567890",
});

> ### Syntax

Reviewer: Style issue not related to this PR, but if you don't separate headings from code blocks with a paragraph/lead-in sentence it renders without enough space between and looks bad:

![Screenshot 2024-09-19 at 8 32 55 AM](https://github.com/user-attachments/assets/cc3c1c09-c6b0-45f5-bf75-73ef05a04d72)

We should either add a boilerplate lead-in sentence or open a ticket in our backlog to fix the style to increase spacing.

### Syntax

```ts
import { getAttributesForm } from '@/auth/api/getAttributesForm';

getAttributesForm(formCode: string);
```

### Returns

`AttributesFormModel`

### Usage

```ts
getAttributesForm(formCode: "customer_account_create");
```

## getCustomerData

The `getCustomerData` function retrieves customer data based on the required `auth_dropin_user_token` parameter. The function calls the [`customer` query](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/queries/customer/).

### Syntax

```ts
import { getCustomerData } from '@/auth/api/getCustomerData';

getCustomerData(auth_dropin_user_token: "abcdefghi12345");
```

### Returns

`CustomerDataModel`

## getCustomerToken

The `getCustomerToken` function handles the sign-in operation. It requires `userName` and `password` parameters and performs the following actions under the hood:

1. Retrieves the customer token.

1. Fetches customer data using the token.

1. Sets the `auth_dropin_firstname` and `auth_dropin_user_token` cookies.

> 1. Publishes an ACDL event.

Reviewer: ```suggestion
1. Publishes an [Adobe Client Data Layer (ACDL)](https://github.com/adobe/adobe-client-data-layer) event.
```

1. Emits an “authenticated” event.

> You can use the `getCustomerToken` function to build a custom authentication flow that remains fully integrated with other drop-ins.

Reviewer: ```suggestion
You can use the `getCustomerToken` function to build a custom authentication flow that remains fully integrated with other dropins.
```

The function calls the [`generateCustomerToken` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/generate-token/).

### Syntax

```ts
import { getCustomerToken } from '@/auth/api/getCustomerToken';

getCustomerToken(
  {
    email: string,
    password: string,
    handleSetInLineAlertProps: (value?: inLineAlertInterface) => void;
    translations: Record<string, string>;
    onErrorCallback?: (value?: unknown) => void;
  },
);
```

### Returns

`getCustomerTokenProps`

### Usage

```ts
getCustomerToken(email: "abc@example.com", password: "MyBadPassword");
```

## getStoreConfig

The `getStoreConfig` function uses the [`storeConfig` query](https://developer.adobe.com/commerce/webapi/graphql/schema/store/queries/store-config/) to retrieve store configuration data.

### Syntax

```ts
import { getStoreConfig } from '@/auth/api/getStoreConfig';

getStoreConfig();
```

### Returns

`StoreConfigModel`

## requestPasswordResetEmail

> The `requestPasswordResetEmail` function  initiate the process to reset a customer's password. The function calls the [`requestPasswordResetEmail` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/request-password-reset-email/).

Reviewer: ```suggestion
The `requestPasswordResetEmail` function initiates the process of resetting a customer's password. The function calls the [`requestPasswordResetEmail` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/request-password-reset-email/).
```

### Syntax

```ts
import { resendConfirmationEmail } from '@/auth/api/resendConfirmationEmail';

resendConfirmationEmail(customerEmail:string);
```

### Returns

`resendConfirmationEmailResponse`

### Usage

```ts
resendConfirmationEmail(customerEmail:"abc@example.com");
```

## resetPassword

The `resetPassword` function resets a customer's password using the supplied `email`, `resetPasswordToken`, and `newPassword` parameters. The function calls the [`resetPassword` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/reset-password/).

### Syntax

```ts
import { resetPassword } from '@/auth/api/resetPassword';

resetPassword (email: string, resetPasswordToken: string, newPassword: string);
```

### Returns

`ResetPasswordModel`

### Usage

```ts
resetPassword(
    email: "abc@example.com", 
    resetPasswordToken: "gh80pkjGdsPyiXc0sUUXswX1uGN7crUr", 
    newPassword: "ThisNewPasswordIsSomewhatBetter");
```

## revokeCustomerToken

The `revokeCustomerToken` function revokes the customer's token and clears cookie. It then publishes an ACDL event and emits an "authenticated" event.

> This API can also be used to build a custom sign-out flow that stays fully integrated with other drop-ins.

Reviewer: ```suggestion
This API can also be used to build a custom sign-out flow that stays fully integrated with other dropins.
```

---
title: User Auth Functions
description: Learn about the API functions provided by the User Auth dropin.
sidebar:
  # label: API
  order: 5
---

import OptionsTable from '@components/OptionsTable.astro';

## confirmEmail

The `confirmEmail` function uses the supplied `customerEmail` and `customerConfirmationKey` parameters to complete the customer activation process. Adobe Commerce sends the confirmation key to the customer when they request to create an account. The function calls the [`confirmEmail` mutation](https://developer.adobe.com/commerce/webapi/graphql/schema/customer/mutations/confirm-email/).

### Signature

```ts
confirmEmail(
  {
    customerEmail: string,
    customerConfirmationKey: string,
  },
);
```

<OptionsTable
  compact
  options={[
    ['Parameter', 'Type', 'Req?', 'Description'],

>     ['customerEmail', 'string', 'Yes', 'The email address to be confirmed..'],

Reviewer: ```suggestion
    ['customerEmail', 'string', 'Yes', 'The email address to be confirmed.'],
```
